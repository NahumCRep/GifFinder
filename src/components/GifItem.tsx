import { GifData } from '../types'

interface Props {
    gif: GifData
}

const GifItem = ({gif}: Props) => {
    return (
        <div className='card'>
            <img src={gif.url} alt={gif.title} />
        </div>
    )
}

export default GifItem