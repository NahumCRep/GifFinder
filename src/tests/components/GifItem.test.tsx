import { render, screen } from "@testing-library/react";
import GifItem from "../../components/GifItem";

describe('Test for GifItem component', () => { 

    const testGifData = {
        id: 'asd12d',
        title: 'Saitama',
        url: 'http://one-punch.org/saitama.jpg'
    }

    test('should match snaptshot', () => {
        const { container } = render(<GifItem gif={testGifData} />)
        expect(container).toMatchSnapshot()
    })

    test('should show the image with the src and alt attributes', () => {
        render(<GifItem gif={testGifData} />)
        const {src, alt} = screen.getByRole('img') as HTMLImageElement
        expect(src).toBe(testGifData.url)
        expect(alt).toBe(testGifData.title)
    })
})