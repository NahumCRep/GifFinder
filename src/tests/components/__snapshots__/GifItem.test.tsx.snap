// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for GifItem component should match snaptshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://one-punch.org/saitama.jpg"
    />
  </div>
</div>
`;
